presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -110,6 +128,7 @@ public HivePartitionManager(
         checkArgument(domainCompactionThreshold >= 1, "domainCompactionThreshold must be at least 1");
         this.domainCompactionThreshold = domainCompactionThreshold;
         this.partitionFilteringFromMetastoreEnabled = partitionFilteringFromMetastoreEnabled;
+        this.executorService = listeningDecorator(newCachedThreadPool(daemonThreadsNamed("partition-value-extractor-%s")));
```

Comment:
Can we bound the thread pool size. Perhaps the boolean parsing is enabled thing can become the number of threads ?

Reply:
Done. bounded it to 100...and also made it configurable